### PR TITLE
POOL 817 - Remainder of work for Reference app

### PR DESCRIPTION
**Pull request by ddesrosier**
- Proper versioning based on `current-pool-data`
- Withdraw bug fix
- Awarding functionality

### CODE REVIEW

File: lib/components/NumOfWinnersControlCard.jsx [hunk 1]
```diff
@@ -40,6 +41,10 @@ const handleSetNumberOfWinners = async (
 }
 
 export const NumOfWinnersControlCard = (props) => {
+  const [contractVersions] = useAtom(contractVersionsAtom)
+
+  if (contractVersions.prizeStrategy.contract === 'SingleRandomWinner') return null
```
Comment (chuckbergeron) [inline lib/components/NumOfWinnersControlCard.jsx:46]: I think we should still show this card but just say 1 for Num Winners.

File: lib/components/WithdrawUI.jsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const handleWithdrawSubmit = async (
   // TX overrides
   params.push({ gasLimit: 800000 })
 
-  await sendTx(setTx, provider, contractAddress, CompoundPrizePoolAbi, method, params, 'Withdraw')
+  await sendTx(setTx, provider, contractAddress, PrizePoolAbi, method, params, 'Withdraw')
```
Comment (chuckbergeron) [inline lib/components/WithdrawUI.jsx:47]: Are these ABIs interchangeable between 3.0.1, 3.1.0 and stake vs. compound prize pool?

**Review by chuckbergeron — CHANGES_REQUESTED** [review]
👍

File: lib/components/WithdrawUI.jsx (hunk 2 above)
Comment (ddesrosier) [reply lib/components/WithdrawUI.jsx:47]: As far as I can tell, yes. All of the methods we use here (and other places I switched it) are implemented on the PrizePool contract, not the individual compound or staking contracts.

**Review by chuckbergeron — APPROVED** [review]

**Pull request merged**